pynetdicom/dimse_messages.py

```diff
@@ -310,14 +330,45 @@ def decode_msg(self, primitive):
                         #   has been completely decoded
                         return True
 
+                    if (
+                        _config.STORE_RECV_CHUNKED_DATASET and
+                        isinstance(self, C_STORE_RQ)
+                    ):
+                        self._data_set_file = NamedTemporaryFile(mode="wb", suffix=".dcm")
+                        self._data_set_path = Path(self._data_set_file.name)
+
+                        from pynetdicom import PYNETDICOM_IMPLEMENTATION_UID
+                        from pynetdicom import PYNETDICOM_IMPLEMENTATION_VERSION
+
+                        self._data_set_file.write(b'\x00' * 128)
+                        self._data_set_file.write(b'DICM')
+
+                        file_meta = Dataset()
+
+                        file_meta.FileMetaInformationGroupLength = 0
+                        file_meta.FileMetaInformationVersion = b'\x00\x01'
+                        file_meta.MediaStorageSOPClassUID = self.command_set.AffectedSOPClassUID
+                        file_meta.MediaStorageSOPInstanceUID = self.command_set.AffectedSOPInstanceUID
+                        file_meta.TransferSyntaxUID = assoc._accepted_cx[context_id].transfer_syntax[0]
```

Comment:
Explicitly check for `assoc = None`

Reply:
Maybe? I mostly made this optional so we wouldn't have to pass an association in the DIMSE message tests. In practice, its passed unconditionally (didn't think it made sense to branch on an unlikely case in the caller). We could also assert in this branch - I'll defer to what you prefer here.

Reply:
>Explicitly check for assoc = None

No, don't worry about it.